File: api/src/main/java/rocks/milspecsg/msreferral/api/member/repository/MongoMemberRepository.java
package rocks.milspecsg.msreferral.api.member.repository;

import org.bson.types.ObjectId;
import org.mongodb.morphia.Datastore;
import org.mongodb.morphia.query.Query;
import rocks.milspecsg.msreferral.model.core.member.Member;

import java.util.Optional;
import java.util.UUID;

public interface MongoMemberRepository extends MemberRepository<ObjectId, Datastore> {

    Optional<? extends Query<Member<ObjectId>>> asQuery(UUID userUUID);
}

File: referral-api/src/main/kotlin/org/anvilpowered/referral/api/member/MongoMemberRepository.kt
/*
 *   Referral - AnvilPowered
 *   Copyright (C) 2020
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Lesser General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU Lesser General Public License for more details.
 *
 *     You should have received a copy of the GNU Lesser General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package org.anvilpowered.referral.api.member

import dev.morphia.query.Query
import org.anvilpowered.referral.api.model.member.Member
import org.bson.types.ObjectId
import java.util.UUID

interface MongoMemberRepository {

    fun asQuery(uuid: UUID): Query<Member<ObjectId>>
}
